### PR TITLE
deps: require sqlalchemy for ibis SQL backend

### PR DESCRIPTION
**Pull request by tswast**

### CODE REVIEW

**Comment by cpcloud** [comment]
I can help with the failing tests here. We recently removed the difference in compilation between `ibis.ifelse` and `ibis.where`. Both are now using the `ops.Where` operation which compiles to an `if` call.

**Comment by tswast** [comment]
Thanks for the help @cpcloud !

**Comment by cpcloud** [comment]
@tswast I think this PR is ready for review if you don't mind!

**Comment by cpcloud** [comment]
It sort of ballooned into a bit of a CI cleanup due to having to futz with poetry-dynamic-versioning.

File: tests/system/test_compiler.py [hunk 1]
```diff
@@ -129,9 +136,8 @@ def test_window_function(alltypes, project_id, dataset_id):
     assert result == expected
 
 
+@pytest.mark.skipif(IBIS_VERSION <= IBIS_1_VERSION, reason="requires ibis 2.x")
```
Comment (tswast) [inline tests/system/test_compiler.py:139]: Nit: I think we can remove this, as we dropped ibis 1.0 support in this package already.

File: tests/system/test_compiler.py [hunk 2]
```diff
@@ -245,20 +249,54 @@ def test_bool_reducers(alltypes, project_id, dataset_id):
     assert result == expected
 
 
-def test_bool_reducers_where(alltypes, project_id, dataset_id):
+@pytest.mark.parametrize(
```
Comment (tswast) [inline tests/system/test_compiler.py:252]: Love this pattern. Thanks for coming up with it!

**Review by tswast — COMMENTED** [review]
LGTM. Thanks so much for all the help!

**Pull request merged**